multi-agent/lib/mas/sharing/cloner.py

```
@@ -244,6 +245,17 @@ def _compute_closure(self, root_rids: Set[str], ctx: CloneContext) -> Dict[str,
                 # Load and validate resource
                 doc = self.resources.get(rid)
 
+                if doc.ownership == ResourceOwnership.BUILTIN:
+                    if doc.visibility != ResourceVisibility.PUBLIC:
+                        raise ValueError(
+                            f"Cannot share: resource {rid} ({doc.name}) is a "
+                            f"draft built-in and would be invisible to the recipient"
+                        )
```

Ruff (0.16.0)

[warning] Abstract `raise` to an inner function

(TRY301)